### PR TITLE
Refactor/better ts typings

### PR DESCRIPTION
**Pull request by BjornMelgaard**
fix https://github.com/char0n/ramda-adjunct/issues/137 + more

commits are self-explanatory

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -210,12 +212,12 @@ declare namespace RamdaAdjunct {
         /**
          * Checks if value is a positive `Number` primitive or object.
          */
-        isPositive(val: any): boolean
+        isPositive(val: any): val is number
```
Comment (char0n) [inline src/index.d.ts]: Are we sure here ? `isPositive` returns false for the NaN, -Infinity, +Infinity. Nevertheless they are still valid `number` values. The same applies for `isNegative`, `isInteger`, `isFloat`

File: src/index.d.ts [hunk 2]
```diff
@@ -468,8 +469,8 @@ declare namespace RamdaAdjunct {
         /**
          * Returns whether or not an object has an own property with the specified name at a given path.
          */
-        hasPath(path: Array<String|number>, obj: any): boolean;
-        hasPath(path: Array<String|number>): (obj: Object) => boolean;
+        hasPath(path: Array<String|number>, obj: object): boolean;
```
Comment (char0n) [inline src/index.d.ts]: String -> string

File: src/index.d.ts (hunk 1 above)
Comment (BjornMelgaard) [reply src/index.d.ts]: I dont know how to narrow number type to omit named values

stackoverflow too
https://stackoverflow.com/questions/46389560/how-to-narrow-number-type-to-omit-nan-infinity-infinity

Comment (BjornMelgaard) [reply src/index.d.ts]: even haskell cant do this I think :smile: (but idris can)

**Comment by BjornMelgaard** [comment]
@char0n  done

**Pull request merged**